apps/aecore/src/aec_peers.erl

```diff
@@ -237,164 +235,146 @@ check_ping_interval_env() ->
 %%% gen_server functions
 %%%=============================================================================
 
--record(state, {peers :: gb_trees:tree(binary(),peer()),
-                blocked = gb_sets:new() :: gb_sets:set(uri()),
-                errored = gb_sets:new() :: gb_sets:set(uri()),
-                local_peer_uri :: uri(),
-                local_peer_host :: string() | undefined,
-                local_peer_port :: integer() | undefined}).
+-record(state, {peers :: gb_trees:tree(binary(), peer()),
+                blocked = gb_sets:new() :: gb_sets:set(http_uri:uri()),
+                errored = gb_sets:new() :: gb_sets:set(http_uri:uri()),
+                local_peer :: peer()}).
```

Comment:
Ok, perhaps at least a comment about which attributes of the #peer{} record that are relevant for `local_peer`? I believe it's just the uri-related parts...

Reply:
Yes, that's right, at the moment we use scheme, host and port. But in the future possibly also the path. The gain on using this type is that we can re-use the universal encoding and decoding of URIs.